Feature/12 save boards to db

joe-riley opened the pull request:
working, I hope :/

joe-riley committed:
Saving for merge

joe-riley committed:
Merge conflicts

Commit:
merging develop

Commit:
merged

Commit:
please code actually save my changes to the repo

Commit:
saving for posterity

Commit:
Merge branch 'develop' into feature/12-save-boards-to-db

Commit:
Change seeds to match code, update board route, be able to save a board.

Commit:
Fix bug where drawing elements are not rendered yet

Commit:
Add user to js to put login, logout and signup

Commit:
Moving seeds into db since they only have to do with the dB

Commit:
Moving seeds into db since they only have to do with the dB

Commit:
Register and login working

Commit:
Validate duplicate username on register

Commit:
Saving to get develop in

Commit:
Saving to get develop in

Commit:
Merge conflicts

Commit:
Tylor thinks he is funny

Commit:
Add ability to save a board

Commit:
SAving to work on deploy

Commit:
Closer to getting profile working

Commit:
Saving for posterity

Commit:
Those paths are really messing with me

Commit:
Now with some gusto

Commit:
replace ,

Commit:
saving for posterity

Commit:
revert to something that works

Commit:
fixed bug with profiles and sessions

Commit:
can get to profiles

kristinmoon commented on controllers/api/board-routes.js:
```diff
@@ -5,21 +5,21 @@ const sequelize = require('../../config/connection');
     include: [
       {
         model: User,
-        attributes: ['username']
```
perhaps this change is why the username isn't pulling/displaying with the boards?

joe-riley replied:
I don't think I changed that. I had changed it to id for a while but I am pretty sure that is how I found it.

joe-riley replied:
This did lead me to another place where I changed the attribute to id through, in home-routes and I that fixes the bug you are referring to. Good catch. It was like a behind the back catch :P

kristinmoon commented on controllers/html/home-routes.js:
```diff
@@ -1,86 +1,77 @@
+    });
+});
+
+module.exports = router;
```
newline needed

joe-riley replied:
done!

kristinmoon commented on db/seeds/board-seeds.js:
```diff
@@ -1,53 +1,53 @@
     title: 'Far far away',
     description: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.Aenean commodo ligula eget dolor.Aenean massa.Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem.Nulla consequat massa quis enim.Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu.In enim justo, rhoncus ut, ...
-    // board_content: insert JSON or string here
+    board_content: '{"some":"Content"}',
```
Is there a dummy "saved" board we can add in here instead of the image to see how this works?

Commit:
Resolving some pr comments

Commit:
Fix username not showing on boards on homepage

kristinmoon commented on public/js/drawing.js:
```diff
@@ -6,13 +6,14 @@
     return document.getElementById(id);
   };
 
-  // var canvas = (this.__canvas = new fabric.Canvas("c", {
```
Perhaps some of these changes toggled the drawing mode on automatically? It was working prior to this...

joe-riley replied:
Crazy, I tried just copieing what is in develop right now and the only thing that works is draw mode AND it doesn't care about line size or color. :/ I will have to work my way through this one

kristinmoon replied:
I just checked out to develop and the shapes and drawing tools are working correctly... maybe something else in another file got changed?

joe-riley replied:
board controller and drawing are the only things that I can think of.

joe-riley replied:
trying...

kristinmoon reviewed (changes requested):
I'm going to be honest, there are SO many changes in this one PR that it's very challenging and time consuming to make individual file/line comments. I wish these commits and pushes would have been broken up into multiple pull requests per ticket or issue that was addressed. I will try my best to make individual comments, but running the app on my browser, I have the following bugs:
- when you click the profile button on any page _except_ for the home page, the link doesn't work. It adds /profile to whatever page you are currently on.
- the delete, save, and clear buttons are missing on the /board/:id pages
- drawing mode is always toggled on somewhere. When you add a shape, you can no longer select it and move it, or resize
- username is no longer pulled with the board data
- you get a post error when you try to save a board even though, a new board id is created. Perhaps because it's not saving the actual board_content data?

joe-riley commented:
> ~~when you click the profile button on any page except for the home page, the link doesn't work. It adds /profile to whatever page you are currently on.~~
- FIXED

> ~~the delete, save, and clear buttons are missing on the /board/:id pages~~
- ~~I think we should talk about this. I think that only the user that owns the board should be able to perform these actions and only on the create/edit board page. Granted, we dont' have the concept of editing an existing board yet but that goes beyond the scope of the two tickets I worked on.~~
- - Created #47 for this

> ~~drawing mode is always toggled on somewhere. When you add a shape, you can no longer select it and move it, or resize~~
- ~~I remember seeing this work in the demo but it never worked right for me. Still looking into it.~~
- FIXED (as long as you add shapes before you draw (this happens in develop too from what I saw)

> ~~username is no longer pulled with the board data~~
- FIXED

> ~~you get a post error when you try to save a board even though, a new board id is created. Perhaps because it's not saving the actual board_content data?~~
- ~~once you click through the error the board is saved. Looking into it and trying to see if I am not awaiting  correctly.~~
- FIXED

Commit:
Fix internal server error when saving board

kristinmoon commented:
> > the delete, save, and clear buttons are missing on the /board/:id pages
> 
> * I think we should talk about this. I think that only the user that owns the board should be able to perform these actions and only on the create/edit board page. Granted, we dont' have the concept of editing an existing board yet but that goes beyond the scope of the two tickets I worked on.

**- Got it! Your comment above makes sense... I was thinking more along the lines of when I click on my own saved board, how do I delete it if I don't want it any more (or clear it or resave)**

JOE: I think this will be a new feature that I will try to cram in but I am trying to work through these errors atm. - - Created #47 for this
> 
> 
> > you get a post error when you try to save a board even though, a new board id is created. Perhaps because it's not saving the actual board_content data?
> 
> * once you click through the error the board is saved. Looking into it and trying to see if I am not awaiting  correctly.

**- It looked to me that the title and description were saved, but not the text or JSON of the actual board rendering itself... When I use Insomnia to GET http://localhost:3001/boards it returns a sequelize database error.**

JOE: Updated my original reply and have fixed that. I am seeing the same error for get boards, which seems to be a different issue all together. Attempting to fix.

Commit:
Restore board controlls, as long as you add shapes before you draw

Commit:
Restore board controlls, as long as you add shapes before you draw

joe-riley commented:
@kristinmoon and or @Zapblumar - 

Please review my fixes. Note that one on the list has a new issue created. I think it is more important to move forward so that I am not stalling more work from being done for any longer.

kristinmoon commented:
Looking good... what did you do to pull the username onto the profile and single boards pages? This also needs to be done for the homepage boards.

For the adding shapes before you draw, you just have to turn off drawing mode when you're done drawing and you can continue to edit the shapes. Perhaps we need some instructions on this page. I can work on that.

Not something for this PR, but for the next one. If we aren't going to let people edit boards, we should remove the drawing tools as they don't currently work on a pre-existing board.

kristinmoon committed:
add format_date to created_at on single-board.hbs

joe-riley commented:
omg you are holding my PR hostage. 

The names show up correctly for me on profile. I don't even touch the single board pages but lemme check.. the names show up just fine for me :/

kristinmoon commented:
idk what to do. I'll approve this and hopefully in the next merge they will magically show up again
